servers/visual/rasterizer.h

```diff
@@ -1256,6 +1272,32 @@ class RasterizerCanvas {
 class Rasterizer {
 protected:
 	static Rasterizer *(*_create_func)();
+	static bool khr_debug_enabled;
+
+	//Profiling Information
+	//Labels
+	static void (*push_label_func)(unsigned int source, unsigned int id, int length, const char *message);
+	static void (*pop_label_func)();
+
+	//Timing stuff
+	static const int FRAME_TIMING_COUNT = 3;
+	static const int MAX_QUERIES = 128;
+
+	static bool frame_timing_enabled;
+	struct FrameTimings {
+		unsigned int queries[MAX_QUERIES];
+		uint32_t result_count;
+		uint32_t query_count;
+		LocalVector<uint64_t> query_results;
+		LocalVector<String> query_names;
+	};
+	FrameTimings frame_timings[FRAME_TIMING_COUNT];
+	int frame = 0;
+
+	static void (*gen_queries_func)(int n, unsigned int *ids);
+	static void (*del_queries_func)(int n, unsigned int *ids);
+	static void (*query_counter_func)(unsigned int id, unsigned int target);
+	static void (*get_query_func)(unsigned int id, unsigned int pname, uint64_t *params);
 
```

Comment:
I'm wondering if this (or some of it) could go in a nested class? Just to make it clear the separation of what is part of `Rasterizer`, and what is to do with the debug timings? (I can't say for definite it would work, sometimes there are restrictions on nested classes.) :thinking: